salvo/src/lib/builder/test_envoy_builder.py

```diff
@@ -12,25 +12,120 @@
 
 logging.basicConfig(level=logging.DEBUG)
 
-def test_build_envoy_image_from_source():
+def _check_call_side_effect(args, parameters):
+  """Examine the incoming arguments for command execution and return the
+  expected or unexpected output.
+
+  Args:
+    args: The arguments supplied to the mocked function
+    parameter: The CommandParameters passed to the cmd_exec method
+  Return:
+    usually this returns a string containing the command output.  In
+      some cases we may raise an exception.
+  """
+
+  assert 'cwd' in parameters._asdict()
+
+  if args == "bazel clean":
+    return "INFO: Starting clean"
+  elif args == "bazel build -c opt " + constants.ENVOY_BINARY_BUILD_TARGET:
+    return "foo"
+  elif args == ("cp -fv bazel-bin/source/exe/envoy-static "
+                "build_release_stripped/envoy"):
+    return "copied..."
+  elif args == (("objcopy --strip-debug bazel-bin/source/exe/envoy-static "
+                 "build_release_stripped/envoy")):
+    return "stripped..."
+  elif args == (("docker build -f ci/Dockerfile-envoy -t "
+                 "envoyproxy/envoy-dev:v1.16.0 --build-arg "
+                 "TARGETPLATFORM='.' .")):
+    return "docker build output..."
+
+  raise NotImplementedError(f"Unhandled arguments in call: {args}")
+
+@mock.patch.object(envoy_builder.EnvoyBuilder, 'create_docker_image')
```

Comment:
(possibly non-actionable) To be noted that the test written like this only serves as a change detector. Everything the test interacts with is a mock, so it only verifies about assumptions about the mocked objects and has no functional value.

A change detector, because anytime we change the code, we have to come here and make the very same change to the mocks. One way to address this would be to decrease the number of mocks in use, maybe at least some of the mocked objects could be real objects performing something on a fake filesystem, etc.

Feel free to just ack if that isn't feasible.

Reply:
I was able to remove the envoy builder mock.  It did feel grimy but I didn't circle back and take a fresh look.

My intent with the mocks was to reduce the usage and size of the side effect methods.

Side note, I did get my IDE's linter working again so you'll see other changs that it's picking up.